prov/efa/src/efa_hmem.c

```diff
@@ -298,6 +298,49 @@ static int efa_hmem_info_init_synapseai(struct efa_hmem_info *synapseai_info, st
 	return 0;
 }
 
+/**
+ * @brief  Validate an FI_OPT_FI_HMEM_P2P (FI_OPT_ENDPOINT) option for a
+ *         specified HMEM interface
+ *
+ * @param  domain  The efa_domain struct which contains an efa_hmem_info array
+ * @param  iface   The fi_hmem_iface enum of the HMEM interface to validate
+ * @param  p2p_opt The P2P option to validate
+ *
+ * @return FI_SUCCESS if the P2P option is valid for the given interface
+ *         FI_OPNOTSUPP if the P2P option is invalid
+ *         FI_ENODATA if the given HMEM interface was not initialized
+ *         FI_EINVAL if p2p_opt is not a valid FI_OPT_FI_HMEM_P2P option
```

Comment:
return value should be negative for an error

Reply:
Ack